app/controllers/templates_controller.rb

```diff
@@ -79,15 +79,21 @@ def auto_complete_controller_name
   end
 
   def preview
-    base = @template.preview_host_collection
-    @host = params[:preview_host_id].present? ? base.find(params[:preview_host_id]) : base.first
+    if params[:id]
+      find_resource
+      base = @template.preview_host_collection
+      @host = params[:preview_host_id].present? ? base.find(params[:preview_host_id]) : base.first
+    else
+      type = request.original_url.split("/")[4]
+      @host = Host.find(params[:preview_host_id])
+      @template = resource_class.new(params[type])
```

Comment:
@eLobato  If Ptable never contain ERB is there a real reason for a preview option in its view?
I have added the type because it was needed deeper in this process, ill research it some more to see if its really needed.